From c45e79040eb595f0959dfaa6d20f615296894d7f Mon Sep 17 00:00:00 2001
From: Colin Cross <ccross@android.com>
Date: Thu, 7 Nov 2013 13:19:34 -0800
Subject: [PATCH 296/634] net: move PPPoLAC and PPPoPNS headers to uapi

Move the entire contents of the linux/if_pppolac.h and
linux/if_pppopns.h headers to uapi, they only contain userspace
interfaces.

Change-Id: I3cfed7f2ae400b53269a1f59144aa3dbc30ae0b5
Signed-off-by: Colin Cross <ccross@android.com>
---
 include/linux/if_pppolac.h      | 12 +-----------
 include/linux/if_pppopns.h      | 11 +----------
 include/uapi/linux/if_pppolac.h | 33 +++++++++++++++++++++++++++++++++
 include/uapi/linux/if_pppopns.h | 32 ++++++++++++++++++++++++++++++++
 4 files changed, 67 insertions(+), 21 deletions(-)
 create mode 100644 include/uapi/linux/if_pppolac.h
 create mode 100644 include/uapi/linux/if_pppopns.h

diff --git a/include/linux/if_pppolac.h b/include/linux/if_pppolac.h
index c06bd6c..e40aa10 100644
--- a/include/linux/if_pppolac.h
+++ b/include/linux/if_pppolac.h
@@ -18,16 +18,6 @@
 #ifndef __LINUX_IF_PPPOLAC_H
 #define __LINUX_IF_PPPOLAC_H
 
-#include <linux/socket.h>
-#include <linux/types.h>
-
-struct sockaddr_pppolac {
-	sa_family_t	sa_family;	/* AF_PPPOX */
-	unsigned int	sa_protocol;	/* PX_PROTO_OLAC */
-	int		udp_socket;
-	struct __attribute__((packed)) {
-		__u16	tunnel, session;
-	} local, remote;
-} __attribute__((packed));
+#include <uapi/linux/if_pppolac.h>
 
 #endif /* __LINUX_IF_PPPOLAC_H */
diff --git a/include/linux/if_pppopns.h b/include/linux/if_pppopns.h
index 0cf34b4..4ac621a9 100644
--- a/include/linux/if_pppopns.h
+++ b/include/linux/if_pppopns.h
@@ -18,15 +18,6 @@
 #ifndef __LINUX_IF_PPPOPNS_H
 #define __LINUX_IF_PPPOPNS_H
 
-#include <linux/socket.h>
-#include <linux/types.h>
-
-struct sockaddr_pppopns {
-	sa_family_t	sa_family;	/* AF_PPPOX */
-	unsigned int	sa_protocol;	/* PX_PROTO_OPNS */
-	int		tcp_socket;
-	__u16		local;
-	__u16		remote;
-} __attribute__((packed));
+#include <uapi/linux/if_pppopns.h>
 
 #endif /* __LINUX_IF_PPPOPNS_H */
diff --git a/include/uapi/linux/if_pppolac.h b/include/uapi/linux/if_pppolac.h
new file mode 100644
index 0000000..b7eb8153
--- /dev/null
+++ b/include/uapi/linux/if_pppolac.h
@@ -0,0 +1,33 @@
+/* include/uapi/linux/if_pppolac.h
+ *
+ * Header for PPP on L2TP Access Concentrator / PPPoLAC Socket (RFC 2661)
+ *
+ * Copyright (C) 2009 Google, Inc.
+ * Author: Chia-chi Yeh <chiachi@android.com>
+ *
+ * This software is licensed under the terms of the GNU General Public
+ * License version 2, as published by the Free Software Foundation, and
+ * may be copied, distributed, and modified under those terms.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ */
+
+#ifndef _UAPI_LINUX_IF_PPPOLAC_H
+#define _UAPI_LINUX_IF_PPPOLAC_H
+
+#include <linux/socket.h>
+#include <linux/types.h>
+
+struct sockaddr_pppolac {
+	sa_family_t	sa_family;	/* AF_PPPOX */
+	unsigned int	sa_protocol;	/* PX_PROTO_OLAC */
+	int		udp_socket;
+	struct __attribute__((packed)) {
+		__u16	tunnel, session;
+	} local, remote;
+} __attribute__((packed));
+
+#endif /* _UAPI_LINUX_IF_PPPOLAC_H */
diff --git a/include/uapi/linux/if_pppopns.h b/include/uapi/linux/if_pppopns.h
new file mode 100644
index 0000000..a392b52
--- /dev/null
+++ b/include/uapi/linux/if_pppopns.h
@@ -0,0 +1,32 @@
+/* include/uapi/linux/if_pppopns.h
+ *
+ * Header for PPP on PPTP Network Server / PPPoPNS Socket (RFC 2637)
+ *
+ * Copyright (C) 2009 Google, Inc.
+ * Author: Chia-chi Yeh <chiachi@android.com>
+ *
+ * This software is licensed under the terms of the GNU General Public
+ * License version 2, as published by the Free Software Foundation, and
+ * may be copied, distributed, and modified under those terms.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ */
+
+#ifndef _UAPI_LINUX_IF_PPPOPNS_H
+#define _UAPI_LINUX_IF_PPPOPNS_H
+
+#include <linux/socket.h>
+#include <linux/types.h>
+
+struct sockaddr_pppopns {
+	sa_family_t	sa_family;	/* AF_PPPOX */
+	unsigned int	sa_protocol;	/* PX_PROTO_OPNS */
+	int		tcp_socket;
+	__u16		local;
+	__u16		remote;
+} __attribute__((packed));
+
+#endif /* _UAPI_LINUX_IF_PPPOPNS_H */
-- 
1.9.1

